subprojects/launcher/src/integTest/groovy/org/gradle/launcher/cli/TaskOptionsSpec.groovy

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.launcher.cli
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class TaskOptionsSpec extends AbstractIntegrationSpec {
+
+    def defineTaskWithProfileOption() {
+        return """
+                abstract class MyTask extends DefaultTask {
+                    @Option(option="profile", description="foobar")
+                    @Optional
+                    @Input
+                    abstract Property<String> getProfile()
+
+                    @TaskAction
+                    void run() {
+                        if (getProfile().isPresent()) {
+                            logger.lifecycle(getName() + "profile=" + getProfile().get())
+                        }
+                    }
+                }
+                """
+    }
+
+    def "will prioritize built-in option over a task option with a conflicting name"() {
```

Comment:
You should add negative conditions, at least for these first 2 basic tests, to ensure that both the built-in task and the custom task option didn't both run.

Just check `!output.contains("<UNWANTED OPTION>")`